src/game/server/gamecontroller.h

```diff
@@ -119,6 +120,32 @@ class IGameController
 
 	void UpdateGameInfo(int ClientID);
 
+	typedef void (*COMMAND_CALLBACK)(class CPlayer *pPlayer, const char *pArgs);
+
+	struct CChatCommand 
+	{
+		const char *m_pName;
+		const char *m_pArgsFormat;
+		const char *m_pHelpText;
+		COMMAND_CALLBACK m_Callback;
+	};
+
+	class CChatCommands
+	{
+		array<CChatCommand*> m_aCommands;
```

Comment:
the check is already there

Reply:
Not that it matters too much, but I think it should be a `if(m_aCommands.size() > MAX_COMMANDS)`

Reply:
Yeah, would be wetter to use some static array for it. Also on client side. Would provide better memory management for the comment @LordSk mentioned.
Another thing: we use `0` instead of `NULL`.

Reply:
@oy my bad, I advised @Ryozuki to replace his `nullptr` with `NULL`s. We have quite a few `NULL` in the code nowadays, isn't there a way to show that it's a pointer type? `0x0`?

Reply:
@Dune-jr no just use 0 for it. The pointer type you show via the naming sheme `p`.
`NULL` is some old c style and when something crazy happens and it isn't 0 you have some nice debugging times ;)

Reply:
Changed NULL to 0.